### DISCUSSION TITLE
Packages / do npm tags work ?

### DISCUSSION THREAD

**Discussion by PeterHewat** [community/community#27112] [Actions]
<p>Hello,</p>
<p>I am using a Github Action to publish to Github Packages (npm):</p>
<p>main.yml:</p>
<pre><code>on: push

jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with:
        node-version: 12
        registry-url: 'https://npm.pkg.github.com'
        scope: '@PeterHewat'
    - run: npm publish # --tag next
      continue-on-error: true
      env:
        NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>If I use the command “npm publish” in my workflow, it creates a package using the current version defined in my packages.json. This works fine. What I do not understand is that if I use the command “npm publish --tag next” in order to specify an npm tag to my package, the package is generated, but I find no trace of the tag. There is a difference though: the package isn’t listed as being the latest (cf. 1.0.5 / 1.0.6):</p>
<p><img src="https://user-images.githubusercontent.com/15655530/181101741-43ac62c6-d2b0-4175-a613-ede9c4d9479d.png" alt="test_github_package_tags.png" data-base62-sha1="gAsYLtDwhM4xl4foR6XM7Kwu5yT" width="419" height="471"></p>
<p>What I would like to do is manage npm tags in order to be able to target latest/next dependencies like what is done here for example:</p>
<p><a href="https://www.npmjs.com/package/@angular/core?activeTab=versions" rel="nofollow noopener">https://www.npmjs.com/package/@angular/core?activeTab=versions</a></p>
<p>Does Github Packages support this?</p>
<p>For now, if I run the command “npm install github:peterhewat/test-packages@next” I get the error “…is not a valid repository name”. I have to remove the tag at the end as follows: “npm install github:peterhewat/test-packages”. This will install the last package created even if it is not specified as “latest version” (ie. 1.0.6 instead of 1.0.5 which is the “latest version”).</p>

**Comment by Yanjingzhu** [comment]
<p>I tried to publish a new version 1.0.1 with next tag to my package, the same as yours, the latest version is still the previous one.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/1b08e34f80a17285b9d05486c15c8edb04239ed2.png" data-download-href="/uploads/short-url/3R9VUkkGQW5p6vBtyqdrstoFQga.png?dl=1" title="next version.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181101743-9c126f54-7364-4a0e-aeab-ef8b3b52f333.png" alt="next version.png" data-base62-sha1="3R9VUkkGQW5p6vBtyqdrstoFQga" width="690" height="310" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/1b08e34f80a17285b9d05486c15c8edb04239ed2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">next version.png</span><span class="informations">786×354 18.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I managed to install package with tag next in the following example: </p>
<pre><code>npm install @yanjingzhu/npmpackages@next
</code></pre>
<p><img src="https://user-images.githubusercontent.com/24806478/181101745-47e45460-ef02-47bf-88de-3b8a3e7c13e4.png" alt="npm install.png" data-base62-sha1="fgPr7mMqhqbgkCmaczCehNjI4AY" width="395" height="314"> </p>
<p>There is the result: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8b27e2473d2f36fd94aca620f776930fa126f5cc.png" data-download-href="/uploads/short-url/jR1ONcWhPui052zJWN6MTpHpKeE.png?dl=1" title="nppm install next.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181101746-eb4c1f29-c1a6-422f-b147-5732c0572d89.png" alt="nppm install next.png" data-base62-sha1="jR1ONcWhPui052zJWN6MTpHpKeE" width="690" height="344" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b27e2473d2f36fd94aca620f776930fa126f5cc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">nppm install next.png</span><span class="informations">838×418 26.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Please check whether this could work for you.</p>

**Comment by PeterHewat** [comment]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a> </p>
<p>Thanks for your message. Unfortunetly, it doesn’t work. If you publish another package with a different tag (ex: “beta”), it won’t be taken into account. The dependency <span class="mention">@next</span> will just get the latest package ignoring the “next” and “beta” tags.</p>
<p>I got the following reply from GitHub Support:</p>
<pre><code>Hi Peter,

Thanks for reaching out.

-- Does Github Packages support this?

Publishing a node package using --tag isn't supported, however you could use a
distribution tag instead– Ie. npm dist-tag add @owner/package@x.x.x latest and
then perform the install using npm install @owner/package@latest.

Please let me know if you need something else.

Cheers,

Melanie
</code></pre>
<p>My initial tests with dist-tag haven’t worked either. I’ll keep on looking into this.</p>

**Comment by PeterHewat** [comment]
<p>Hello again <a class="mention" href="/yanjingzhu">@yanjingzhu</a> </p>
<p>–tag does seem to work for my use case after all.</p>
<p>Cheers,</p>
<p>Peter </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>I am so glad to hear that you have resolved your issue. In my example, I also used  <strong>npm publish --tag next</strong> as you said . </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181101749-af1741e7-edd0-4370-9311-f6142fef1d9f.png" alt="next tag.png" data-base62-sha1="qcXopDPGFnEeSE5mhbLdqQBO4rg" width="416" height="441"></p>

**Comment by roel1975** [comment]
<p>One of the last pages of packages handels about combination of automatisation and modification, actions</p>

**Comment by roel1975** [comment]
<p>That I must evaluate first, but yes it’s the only exception with packages. The last example I guess</p>

**Comment by roel1975** [comment]
<p>Good job! I was sure it worked in theory, but when I read about it I was sure that someone was bullying me again. It’s fun if it’s friendly, strange doug that yesterday everyone believed that woman. Then I panicked.</p>

**Comment by roel1975** [comment]
<p>Are you now the issuer? Or am I wrong?</p>

**Comment by roel1975** [comment]
<p>What a pithy. It should be good training stuff.</p>

**Comment by kanziw** [comment]
<p><code>npm publish --tag CUSTOM_TAG</code> is work. but npm dist-tag add <span class="mention">@owner</span>/package@version tag is not work.</p>
<pre><code class="lang-auto">❯ npm dist-tag add @owner/package@$VERSION $TAG
npm ERR! code E404
npm ERR! 404 Not Found - GET https://npm.pkg.github.com/banksalad/-/package/@owner%2fpackage/dist-tags
npm ERR! 404 
npm ERR! 404  '@owner/package@version' is not in the npm registry.
...
</code></pre>
